spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreTemplate.java

```diff
@@ -521,4 +539,67 @@ private DatastoreReaderWriter getDatastoreReadWriter() {
 						.currentTransactionStatus()).getTransaction()).getTransaction()
 				: this.datastore;
 	}
+
+	private <T> StructuredQuery exampleToQuery(Example<T> example, DatastoreQueryOptions queryOptions, boolean keyQuery) {
+		validateExample(example);
+
+		T probe = example.getProbe();
+		FullEntity.Builder<IncompleteKey> probeEntityBuilder = Entity.newBuilder();
+		this.datastoreEntityConverter.write(probe, probeEntityBuilder);
+
+		FullEntity<IncompleteKey> probeEntity = probeEntityBuilder.build();
+		DatastorePersistentEntity<?> persistentEntity =
+				this.datastoreMappingContext.getPersistentEntity(example.getProbeType());
+
+		StructuredQuery.Builder builder = keyQuery ? Query.newKeyQueryBuilder() : Query.newEntityQueryBuilder();
+		builder.setKind(persistentEntity.kindName());
+
+		ExampleMatcherAccessor matcherAccessor = new ExampleMatcherAccessor(example.getMatcher());
+		matcherAccessor.getPropertySpecifiers();
+		LinkedList<StructuredQuery.Filter> filters = new LinkedList<>();
```

Comment:
The following 10 lines or so construct a list of filters and apply them to a query, which is similar to what we do in Part Tree Datastore Query Method. Would it be possible to change the code there slightly and perhaps expose a static method that we can also use here to construct the filters? 

It might be not worth it because this code here isn't that long anyway, but if it can be combined then I think it would be cleaner.

Reply:
Only two lines are common, so I don't think it makes sense